custom_components/test/fuzzing/fuzzer_utils/ValuePoolFuzzer.py

```diff
@@ -5,6 +5,117 @@
+    """
+    Generates ranking based on length of the lists.
+
+    :param lists: A list containing inner lists.
+    :type lists: list of lists
+    
+    :return: A ranking of the lists based on their length.
+    :rtype: list
+    """
+    rank = sorted(range(len(lists)), key=lambda x: (-len(lists[x]), x))
+    return rank
+
+def _get_repeater(lists, param_combi, m):
+    """
+    Calculates how often a list must be repeated if param_combi is greater than 1.
+
+    :param lists: A list containing inner lists.
+    :type lists: list of lists
+    :param param_combi: Number of parameters to be combined.
+    :type param_combi: int
+    :param m: The current index in the lists.
+    :type m: int
+    
+    :return: The number of times a list must be repeated.
+    :rtype: int
+    """
+    repeater = 1
+    param_combi_count = 1
+    while param_combi_count < param_combi:
+        if m >= param_combi_count:
+            repeater = repeater * len(lists[m-1])
+        param_combi_count += 1
+        
+    return repeater-1
+    
+def _generate_new_list(lists, param_combi, rank):
+    """
+    Generates a new list with recombinations of inner lists.
+
+    :param lists: A list containing inner lists of values.
+    :type lists: list of lists
+    :param param_combi: Number of parameters to be combined.
+    :type param_combi: int
+    :param rank: Indices of the lists sorted by their length.
+    :type rank: list
+    
+    :return: A new list with recombinations of the original lists.
+    :rtype: list of lists
+    """
+    new_lists = []
+    spare_list = []
+    new_spare_list = []
+        
+
+    m = 0
+    while m < len(lists):
+        repeater = _get_repeater(lists, param_combi, m)
+        repeater_count = 0
+        spare_list = []
+        new_spare_list = []
+        while repeater_count <= repeater: 
+            n = m + 1
+            repetition_counter = 1
+            while n < param_combi:
+                repetition_counter = repetition_counter * len(lists[rank[n]])
+                n += 1
+                
+                
+            o = 0
+            index = 0
+            while o < len(lists[rank[m]]):
+                r = 0
+                spare_list = lists[rank[m]]
+                q = len(spare_list)
+                while q < len(lists[rank[0]]):
+                    """if the length of a list is smaller than the longest list we have to fill it, it could probably be more elegant"""
+                    spare_list.append(spare_list[0])
+                    q += 1
+                while r < repetition_counter:
+                    new_spare_list.append(spare_list[o])
+                    index += 1
+                    r += 1
+                o += 1
+            p = len(lists[rank[m]])
+                
+            repeater_count += 1
+                
+        new_lists.append(new_spare_list)
+                
+        m += 1
+        
+    return new_lists
+    
+def _generate_combinations(lists, param_combi):
```

Comment:
Why is this function outside the `ValuePoolFuzzer` class? I think it is a private member of the `ValuePoolFuzzer` class.